field_audit/models.py

```diff
@@ -568,13 +573,44 @@ def delete(self, *, audit_action=AuditAction.RAISE):
         return value
 
     @validate_audit_action
-    def update(self, *args, audit_action=AuditAction.RAISE, **kw):
+    def update(self, audit_action=AuditAction.RAISE, **kw):
         if audit_action is AuditAction.IGNORE:
-            return super().update(*args, **kw)
-        else:
-            raise NotImplementedError(
-                "Change auditing is not implemented for update()."
-            )
+            return super().update(**kw)
+        assert audit_action is AuditAction.AUDIT, audit_action
+
+        fields_to_update = set(kw.keys())
+        audited_fields = set(
+            getattr(self.model, AuditEvent.ATTACH_FIELD_NAMES_AT)
+        )
+        fields_to_audit = fields_to_update & audited_fields
+        if not fields_to_audit:
+            # no audited fields are changing
+            return super().update(**kw)
+
+        values_to_fetch = fields_to_update | {"pk"}
+        old_values = {}
+        for value in self.values(*values_to_fetch):
+            pk = value.pop('pk')
+            old_values[pk] = value
+
+        # open a transaction in db of audited model
+        db = self.first()._state.db
```

Comment:
Ideally could determine the db to use without fetching any instances to limit DB hits, but as far as I know we need an actual instance to figure this out, which requires fetching one instance from the queryset.

Reply:
Offlined with Cal who recommended using `django.db.routers.db_for_write` which looks up which db to write to based on model class, meaning we can avoid this unnecessary fetch, and this is the standard approach in Django to determine which DBs to read/write from. It's a win win. See 2206aeea2b71f51e1422ff5656e928a577cd8d8a